app/resources/v1/reception_resource.rb

```diff
@@ -77,7 +77,7 @@ def permitted_request_attributes
     def permitted_sample_attributes
       %i[name external_id species study_uuid priority_level public_name
          sanger_sample_id supplier_name taxon_id donor_id country_of_origin
-         accession_number date_of_sample_collection]
+         accession_number date_of_sample_collection retention_instruction]
```

Comment:
Just a doubt - I hope we don't need a separate validation for retention_instruction here as it will be handled as part of the schema validation for any unexpected values.

Reply:
Avro isn't involved in the creation of the message being processed here, so it could very easily not pass validation.

Reply:
It is, but then you map the values around and so there's currently nothing stopping you submitting anything you want to this Traction endpoint.  Plus the implementation here doesn't know you put it through Avro.  If in future we change the way we create the messages coming into this API, we'd like to have confidence that we can't submit values that cause issues.

Reply:
That is quite true. I understand now what you meant by Avro isn't involved here. 

I'll probably add some validations for the retention instructions. I'm not sure if any other system calls this endpoint but in that case we'll need to have some validations for retention instructions here.